src/main/java/org/folio/rest/support/InstanceUtil.java

```diff
@@ -0,0 +1,269 @@
+  public static final String SOURCE_FIELD = "source";
+  public static final String MATCH_KEY_FIELD = "matchKey";
+  public static final String TITLE_FIELD = "title";
+  public static final String INDEX_TITLE_FIELD = "indexTitle";
+  public static final String ALTERNATIVE_TITLES_FIELD = "alternativeTitles";
+  public static final String EDITIONS_FIELD = "editions";
+  public static final String SERIES_FIELD = "series";
+  public static final String IDENTIFIERS_FIELD = "identifiers";
+  public static final String CONTRIBUTORS_FIELD = "contributors";
+  public static final String SUBJECTS_FIELD = "subjects";
+  public static final String CLASSIFICATIONS_FIELD = "classifications";
+  public static final String PUBLICATION_FIELD = "publication";
+  public static final String PUBLICATION_FREQUENCY_FIELD = "publicationFrequency";
+  public static final String PUBLICATION_RANGE_FIELD = "publicationRange";
+  public static final String PUBLICATION_PERIOD_FIELD = "publicationPeriod";
+  public static final String ELECTRONIC_ACCESS_FIELD = "electronicAccess";
+  public static final String INSTANCE_TYPE_ID_FIELD = "instanceTypeId";
+  public static final String INSTANCE_FORMAT_IDS_FIELD = "instanceFormatIds";
+  public static final String PHYSICAL_DESCRIPTIONS_FIELD = "physicalDescriptions";
+  public static final String LANGUAGES_FIELD = "languages";
+  public static final String NOTES_FIELD = "notes";
+  public static final String ADMINISTRATIVE_NOTES_FIELD = "administrativeNotes";
+  public static final String MODE_OF_ISSUANCE_ID_FIELD = "modeOfIssuanceId";
+  public static final String CATALOGED_DATE_FIELD = "catalogedDate";
+  public static final String PREVIOUSLY_HELD_FIELD = "previouslyHeld";
+  public static final String STAFF_SUPPRESS_FIELD = "staffSuppress";
+  public static final String DISCOVERY_SUPPRESS_FIELD = "discoverySuppress";
+  public static final String STATISTICAL_CODE_IDS_FIELD = "statisticalCodeIds";
+  public static final String SOURCE_RECORD_FORMAT_FIELD = "sourceRecordFormat";
+  public static final String STATUS_ID_FIELD = "statusId";
+  public static final String STATUS_UPDATED_DATE_FIELD = "statusUpdatedDate";
+  public static final String TAGS_FIELD = "tags";
+  public static final String NATURE_OF_CONTENT_TERM_IDS_FIELD = "natureOfContentTermIds";
+  public static final String AUTHORITY_ID_FIELD = "authorityId";
+  public static final String VALUE_FIELD = "value";
+
+  private InstanceUtil() {
+    throw new UnsupportedOperationException("Cannot instantiate utility class");
+  }
+
+  /**
+   * Maps instance json representation that corresponds to the mod-inventory module instance schema to instance object
+   * corresponding to the inventory storage module schema.
+   *
+   * @param instanceDtoJson - mod-inventory instance representation as json object
+   * @return {@link Instance} object
+   */
+  public static Instance mapInstanceDtoJsonToInstance(JsonObject instanceDtoJson) {
+    return new Instance()
```

Comment:
I'm very concerned about this. mod-inventory-storage must not know anything about mod-inventory schema.
What is the reason for that?

Reply:
understand your concern, this is necessary to convert the mapped instances retrieved from a file on S3 to the instance representation used by the mod-inventory-storage module. The mapping logic used in the mod-marc-migration to remap instances produces instances that correspond to the mod-inventory instance schema. In order to provide a clearer understanding about the incoming instance structure, the schema that the incoming instance representation corresponds to is described/mentioned here.
Do you prefer to describe it here smth like: "methods maps bulk instance record representation retrieved from external bulk instances file to inventory storage instance", or do you have another proposal?

Reply:
Hello @RuslanLavrov,

I understand. This class will likely be challenging to maintain. I recommend creating a schema for this instance representation that extends the current module instance schema. This would enable you to use JSON-to-object deserialization, which I believe would greatly streamline the process.

Additionally, looking ahead, you could consider refactoring to introduce a new endpoint that utilizes this extended schema. In mod-inventory, this would simplify operations by allowing you to call this new endpoint directly, eliminating the need for additional calls (for preceding/succeeding titles).

Thoughts?

Reply:
Hello @psmagin, could you elaborate on how adding a new schema would allow to use json-to-object deserialization to inventory-storage instance object? It’s not quite clear because this method maps instance json (that correspond to the mod-inventory schema) to the inventory-storage instance representation. Additionally, during this mapping some fields (e.g. preceding/succeeding titles) need to be ignored. Or do you mean that the method could transfer data to the existing Instance object from instance object representation generated by the new proposed schema instead of from a JsonObject?